datafusion/core/src/physical_plan/filter.rs

```diff
@@ -191,11 +190,8 @@ impl ExecutionPlan for FilterExec {
     fn statistics(&self) -> Statistics {
         let predicate = self.predicate();
 
-        if let Some(binary) = predicate.as_any().downcast_ref::<BinaryExpr>() {
-            let columns = collect_columns(predicate);
-            if !is_operator_supported(binary.op()) || columns.is_empty() {
-                return Statistics::default();
-            }
+        if !check_support(predicate) {
```

Comment:
There is an existing function that does the correctly recursive check

Reply:
Do we need to keep `columns.is_empty()` check?

Reply:
I don't know its purpose -- given that all the tests still pass I think the answer is no, but it would be good to get a second opinion on that

Reply:
I guess it means if the predicate doesn't involve any column, we cannot do selectivity check so just return default?